### ISSUE TITLE
Incorrect .ggpack file (Epic Store)

### ISSUE THREAD

**Issue by kenderipa** [scemino/engge2#5]
Hello, 

I get the following error while starting the game with .ggfiles from Epic Store.

2023-11-05T13:44:55 | INFO | # Welcome to engge II
2023-11-05T13:44:55 | INFO | Host: amd64 / windows
2023-11-05T13:44:55 | INFO | Nim: 1.6.10
2023-11-05T13:44:55 | INFO | Search ggpack in ThimbleweedPark.ggpack*
2023-11-05T13:44:55 | INFO | Add ggpack ThimbleweedPark.ggpack1
ggtable_decoder.nim(92)  newGGTableDecoder
Error: unhandled exception: This is NOT a valid ggpack file :( C443C201 [Exception]

Probably because it contains the DLC already? Could you please take a look?

**Comment by kenderipa** [comment]
ALSA lib dlmisc.c:339:(snd_dlobj_cache_get0) Cannot open shared library libasound_module_pcm_pipewire.so (/lib/x86_64-linux-gnu/alsa-lib/libasound_module_pcm_pipewire.so: cannot open shared object file: No such file or directory)
2023-11-05T13:56:06 | INFO | # Welcome to engge II
2023-11-05T13:56:06 | INFO | Host: amd64 / linux
2023-11-05T13:56:06 | INFO | Nim: 1.6.10
2023-11-05T13:56:06 | INFO | Search ggpack in ThimbleweedPark.ggpack*
2023-11-05T13:56:06 | INFO | Add ggpack ThimbleweedPark.ggpack1
ggtable_decoder.nim(92)  newGGTableDecoder
Error: unhandled exception: This is NOT a valid ggpack file :( C443C201 [Exception]

**Comment by kenderipa** [comment]
Looks like a duplicate.

**Comment by kenderipa** [comment]
I'm reopening this issue, since the solution described in #1 doesn't work.

**Comment by kenderipa** [comment]
@scemino, any suggestions?

**Comment by scemino** [comment]
Did you create a file named `prefs.json`with this content:
```
{
"key": "56ad"
}
```

**Comment by kenderipa** [comment]
> Did you create a file named `prefs.json`with this content:
> 
> ```
> {
> "key": "56ad"
> }
> ```

Yes I did. I also tried all the other keys listed in #1.

**Comment by kenderipa** [comment]
@scemino, any further suggestions?

Also, the engine depends on SDL2-dev. Is this relly needed?

**Comment by scemino** [comment]
I reproduced the issue with the Epic Version, I get this error too: `Error: unhandled exception: This is NOT a valid ggpack file :( C443C201 [Exception]`

Then I created a file next to the `engge2` executable named `Prefs.json` with these contents: 
```json
{
  "key": "566d"
}
```
And it works.

**Comment by kenderipa** [comment]
> I reproduced the issue with the Epic Version, I get this error too: `Error: unhandled exception: This is NOT a valid ggpack file :( C443C201 [Exception]`
> 
> Then I created a file next to the `engge2` executable named `Prefs.json` with these contents:
> 
> ```json
> {
>   "key": "566d"
> }
> ```
> 
> And it works.

Yeah, capitalization. If capitalized, it works. If not, it doesn't. And every single instance of this file name (except readme) weren't capitalized until now. 

Thanks!

**Comment by kenderipa** [comment]
Do we need sdl2-dev, however? My understanding is, it's normally used in development.

**Comment by scemino** [comment]
> Do we need sdl2-dev, however? My understanding is, it's normally used in development.

Did I recommend to install sdl2-dev somewhere ? Why this question ?

This issue is resolved, I can close it, do you agree ?

**Comment by kenderipa** [comment]
> > Do we need sdl2-dev, however? My understanding is, it's normally used in development.
> 
> Did I recommend to install sdl2-dev somewhere ? Why this question ?
> 
> This issue is resolved, I can close it, do you agree ?

Sure. But just so you know, engge2 cannot start without sdl2-dev. And sdl2-dev pulls a lot of packages I don't need :(

**Comment by scemino** [comment]
> > > Do we need sdl2-dev, however? My understanding is, it's normally used in development.
> > 
> > 
> > Did I recommend to install sdl2-dev somewhere ? Why this question ?
> > This issue is resolved, I can close it, do you agree ?
> 
> Sure. But just so you know, engge2 cannot start without sdl2-dev. And sdl2-dev pulls a lot of packages I don't need :(

This is an other subject, feel free to create another issue or send a pull request to fix it.
